customer/src/test/java/nl/softwarestrijders/waiter/customer/domain/AddressTest.java

```diff
@@ -0,0 +1,129 @@
+		assertDoesNotThrow(() -> new Address(houseNumber, addition, "Test street", postalCode, city));
+	}
+
+	@Test
+	@DisplayName("Should throw InvalidHouseNumberException if housenumber is invalid")
+	void shouldThrowWhenInvalidHouseNumber() {
+		assertThrows(InvalidHouseNumberException.class, () -> new Address(- 1, addition, street, postalCode, city));
+	}
+
+	@Test
+	@DisplayName("Should throw if addition is invalid")
+	void shouldTrowWhenInvalidAddition() {
+		assertThrows(InvalidAdditionException.class, () -> new Address(houseNumber, "-", street, postalCode, city));
+	}
+
+	@Test
+	@DisplayName("Should throw if addition contains whitespaces")
+	void shouldNotThrowWhenAdditionContainsWhiteSpace() {
+		assertDoesNotThrow(() -> new Address(houseNumber, "A 4", street, postalCode, city));
+	}
+
+	@Test
+	@DisplayName("Should not throw if street name contains a hyphen")
+	void shouldNotThrowWhenHyphenInStreet() {
+		assertDoesNotThrow(() -> new Address(houseNumber, addition, "Test-Street", postalCode, city));
+	}
+
+	@Test
+	@DisplayName("Should Throw if streetname ends with something else than a number or a letter")
+	void shouldTrowWhenInvalidStreet() {
+		assertThrows(InvalidStreetException.class, () -> new Address(houseNumber, addition, "Test-Street-", postalCode, city));
+	}
+
+	@Test
+	@DisplayName("Should throw when street contains white space")
+	void shouldThrowWhenStreetContainsWhitespace() {
+		assertDoesNotThrow(() -> new Address(houseNumber, addition, "Test Street", postalCode, city));
+	}
+
+	@Test
+	@DisplayName("Should throw when street name contains invalid characters")
+	void shouldThrowWhenStreetContainsInvalidCharacter() {
+		assertThrows(InvalidStreetException.class, () -> new Address(houseNumber, addition, "Test*Street", postalCode, city));
+	}
+
+	@Test
+	@DisplayName("Should throw when Postal code contains too many numbers")
+	void shouldTrowWhenPostalcodeContainsTooManyNumbers() {
+		assertThrows(InvalidPostalCodeException.class, () -> new Address(houseNumber, addition, street, "2211AGA", city));
+	}
+
+	@Test
+	@DisplayName("Should throw when Postal code contains letters in the digit part")
+	void shouldTrowWhenPostalcodeContainsLettersInDigitPart() {
+		assertThrows(InvalidPostalCodeException.class, () -> new Address(houseNumber, addition, street, "2A11AA", city));
+	}
+
+	@Test
+	@DisplayName("Should throw when postal code contains to many letters")
+	void shouldTrowWhenPostalcodeContainsTooManyLetters() {
+		assertThrows(InvalidPostalCodeException.class, () -> new Address(houseNumber, addition, street, "22111A", city));
+	}
+
+	@Test
+	@DisplayName("Should throw if postal code contains invalid characters")
+	void shouldTrowWhenPostalcodeContainsinvalidCharacters() {
+		assertThrows(InvalidPostalCodeException.class, () -> new Address(houseNumber, addition, street, "2211*A", city));
+	}
+
+	@Test
+	@DisplayName("Should throw when city contains invalid characters")
+	void shouldTrowWhenCityContainsInvalidCharacter() {
```

Comment:
Please fix all the grammatical erros of `Trow` -> `Throw` (there are more)

```suggestion
	void shouldThrowWhenCityContainsInvalidCharacter() {
```

Reply:
Oof